flatland/envs/rail_grid_transition_map.py

```diff
@@ -166,20 +131,31 @@ def _check_action_new(self, action: RailEnvActions, position: IntVector2D, direc
             # new_direction will be the only valid transition
             # - take only available transition
             new_direction = fast_argmax(possible_transitions)
-            transition_valid = True
-            return new_direction, transition_valid
+            # TODO generalize / should not be necessary.
+            if action == RailEnvActions.MOVE_LEFT and new_direction != (direction - 1) % 4:
+                action = RailEnvActions.MOVE_FORWARD
+            elif action == RailEnvActions.MOVE_RIGHT and new_direction != (direction + 1) % 4:
+                action = RailEnvActions.MOVE_FORWARD
+            return new_direction, True, action
```

Comment:
MOVE_LEFT/MOVE_RIGHT: turn left/right without acceleration
MOVE_FORWARD: accelerate or swap direction in dead-end with acceleration
DO_NOTHING: move forward or swap direction in dead-end without acceleration